libraries/AP_Math/fourier.cpp

```diff
@@ -0,0 +1,238 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+ 
+ //designed by Philippe Crochat / Anemos Technologies : pcrochat@anemos-technologies.com
+
+#include "fourier.h"
+
+Fourier_Analysis::Fourier_Analysis(){
+	int i;
+	
+	for(i=0;i<2;i++)
+	{
+		_fourier_transform[i].x=0.0f;
+		_fourier_transform[i].y=0.0f;
+	}
+
+	_time_elapsed=0.0f;
+	_buffer_index=0;
```

Comment:
These all init to zero with C++11 so you don't need to define then all here. Just used up flash space because they are gaurenteed to already be zero by the time this constructor is called

Reply:
ok